File: src/impl/InventoryException.cpp
#include "../headers/InventoryException.hpp"
#include "string"

const string InventoryException::msg[] = {
    "Inventory penuh", "Inventory kosong",
    "Item tersebut tidak ada di inventory",
    "Tidak ada item di indeks tersebut"};

InventoryException::InventoryException(int id) : exceptionID(id) {}
const char* InventoryException::what() {
    return msg[this->exceptionID].c_str();
}

File: src/main/java/EngimonHunter2000/InventoryException.java
package EngimonHunter2000;

/**
 * Exception ketika terjadi kesalahan pada {@link Inventory}
 * ID  : Alasan
 * ----|--------
 * 0   : Inventory Penuh 
 * 1   : Inventory Kosong
 * 2   : Item tidak ada di inventory
 * 3   : Indeks item tidak valid
 */
public class InventoryException extends EngimonHunter2000Exception{
  private static final long serialVersionUID = 1l;
  private final int msgID;
  static private final String[] msgs = {
    "Inventory Penuh",
    "Inventory Kosong",
    "Item tidak ada di inventory",
    "Item tidak ditemukan di inventory"
  };

  public InventoryException(int id){
    this.msgID = id;
  }

  public String what(){
    return msgs[this.msgID];
  }

  public void bruh(){
    System.out.println(what());
  }
}
